doc/NAT/SAI-NAT-API.md

```diff
@@ -0,0 +1,443 @@
+
+Per NAT entry stats and hit bit is retrieved by “get_nat_entry_attributes()”.
+##### Step 1: Read SNAT entry hit bit and packet/byte count:
+```sh
+sai_attribute_t nat_entry_attr[10];
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_HIT_BIT;
+
+nat_entry_attr[1].id = SAI_NAT_ENTRY_ATTR_BYTE_COUNT;
+
+nat_entry_attr[2].id = SAI_NAT_ENTRY_ATTR_PACKET_COUNT;
+
+attr_count = 3;
+
+get_nat_entry_attributes(snat_entry, attr_count, nat_entry_attr);
+```
+
+##### Step 2: Read SNAT Zone 100 Translations Done Packet Count:
+```sh
+sai_attribute_t nat_zone_counter_attr[10];
+
+nat_zone_counter_attr[0].id = SAI_NAT_ZONE_COUNTER_ATTR_NAT_TYPE;
+nat_zone_counter _attr[0].value = SAI_NAT_TYPE_SOURCE_NAT;
+
+nat_zone_counter_attr[1].id = SAI_NAT_ZONE_COUNTER_ATTR_ZONE_ID;
+nat_zone_counter_attr[1].value.u32 = 100;
+
+nat_zone_counter_attr[2].id = SAI_NAT_ZONE_COUNTER_ATTR_TRANSLATIONS_PACKET_COUNT;
+
+get_nat_zone_counter_attributes(nat_zone100_counter_id, attr_count, nat_zone_counter_attr);
+```
+
+## 7.0 Error Conditions
+NAT entry is comprised of keys and attributes. Attributes are guarded by run time metadata check with validonly flag. For example following condition states that attribute source IP is valid only if NAT type is SNAT or Double NAT.
+
+```sh
+ /**
+     * @brief Replace source IPv4 address in packet.
+     * NAT actions will be
+     *    (source/destination/both is identified by type of NAT)
+     *    - replace IP address
+     *    - replace layer 4 source port
+     *    - replace layer 4 destination port
+     *
+     * @type sai_ip4_t
+     * @flags CREATE_AND_SET
+     * @default 0.0.0.0
+     * @validonly SAI_NAT_ENTRY_ATTR_NAT_TYPE == SAI_NAT_TYPE_SOURCE_NAT or SAI_NAT_ENTRY_ATTR_NAT_TYPE == SAI_NAT_TYPE_DOUBLE_NAT
+     */
+    SAI_NAT_ENTRY_ATTR_SRC_IP,
+```
+
+Similar check is not possible for the keys in NAT entry given that NAT type is not a match field but a result of a match. To handle such conditions even if SAI API invocator may specify extra keys, SAI driver MUST pick up the correct set of keys based on NAT type in the attribute. For example, in the following example NAT type is DNAT where src_ip and src_mask are present in keys. SAI driver MUST ignore extra keys in such cases.
+
+```sh
+sai_attribute_t nat_entry_attr[10];
+nat_entry_t subnet_nat_entry;
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_NAT_TYPE;
+nat_entry_attr[0].value = SAI_NAT_TYPE_DESTINATION_NAT;
+
+nat_entry_attr[1].id = SAI_NAT_ENTRY_ATTR_DST_IP;
+nat_entry_attr[1].value.u32 = 128.17.18.0; //example string
+
+nat_entry_attr[2].id = SAI_NAT_ENTRY_ATTR_DST_IP_MASK;
+nat_entry_attr[2].value.u32 = 0xffffff00;
+
+nat_entry_attr[3].id = SAI_NAT_ENTRY_ATTR_TO_ZONE;
+nat_entry_attr[3].value.u32 = 200;
+nat_entry_attr[4].id = SAI_NAT_ENTRY_ATTR_FROM_ZONE;
+nat_entry_attr[4].value.u32 =100;

```

Comment:
These attributes no longer exist, should be deleted

Reply:
removed